virtcontainers/container.go

```diff
@@ -566,6 +562,38 @@ func newContainer(sandbox *Sandbox, contConfig ContainerConfig) (*Container, err
 		c.mounts = mounts
 	}
 
+	// iterate all mounts and create block device if it's block based.
+	for i, m := range c.mounts {
+		if len(m.BlockDeviceID) > 0 || m.Type != "bind" {
+			// Non-empty m.BlockDeviceID indicates there's already one device
+			// associated with the mount,so no need to create a new device for it
+			// and we only create block device for bind mount
+			continue
+		}
+
+		var stat unix.Stat_t
+		if err := unix.Stat(m.Source, &stat); err != nil {
+			return nil, fmt.Errorf("stat %q failed: %v", m.Source, err)
+		}
+
+		// Check if mount is a block device file. If it is, the block device will be attached to the host
+		// instead of passing this as a shared mount.
+		if c.checkBlockDeviceSupport() && stat.Mode&unix.S_IFBLK == unix.S_IFBLK {
+			b, err := c.sandbox.devManager.NewDevice(config.DeviceInfo{
+				HostPath:      m.Source,
+				ContainerPath: m.Destination,
+				DevType:       "b",
+				Major:         int64(unix.Major(stat.Rdev)),
```

Comment:
Nits: it seems that `Major` and `Minor` is never used and can be cleaned up entirely. From VM point of view, there is no need to ask for host device major and minor. So it's better to just remove them from the device API. Not required for this PR though.

Reply:
I think `Major` and `Minor` is better option for identifying a device than a path, because we can have  lots of alias, besides the `Major` and `Minor` is from OCI spec but HostPath isn't

Reply:
Yes, I get that. I was thinking to push the burden of converting from <Major, Minor> to a HostPath to the client , because they do not make sense for VMs, and for remove block devices, there is no host device major/minor.  Once that is done, we can remove `Major`/`Minor` from the API.

Reply:
We can discuss more later in seperate PR :-)